host/src/useLocalStorageSync.ts

```diff
@@ -1,17 +1,23 @@
 /* eslint-disable no-unused-vars */
 import React from 'react';
 
-declare global {
-  const localStorage: {
-    getItem: (key: string) => string;
-    setItem: (key: string, object: any) => void;
-  };
-}
+type LocalStorageSetter<T> = {
```

Comment:
I put the base tsconfig root to be `.` rather than `./src` and its giving the correct types for `window` and `localStorage` etc now

Reply:
interesting. I had issues, I think with the type generation, which is why I changed it to `./src`
Great that it's working though, it's much better